LPS-12345

Youness-ELMO opened the pull request.

pedroamarall commented:
ci:test:sf

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>sf</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-37.liferay.com/job/test-portal-source-format/8233/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-1-37.liferay.com/userContent/jobs/test-portal-source-format/builds/8233/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:x: ci:test:sf - 0 out of 1 jobs passed in 5 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/6e716e8827644e1619318501f5d5243dea45fe8b">6e716e8827644e1619318501f5d5243dea45fe8b</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/Youness-ELMO/liferay-portal/tree/LPS-12345-Poshi-exercises">LPS-12345-Poshi-exercises</a><br/>Branch GIT ID: <a href="https://github.com/Youness-ELMO/liferay-portal/commit/68057fd6266d744de5022dc48316cac44759a32a">68057fd6266d744de5022dc48316cac44759a32a</a></p>0 out of 1jobs PASSED<div><h4>1 Failed Jobs:</h4><ul><li><a href="https://test-1-37.liferay.com/job/test-portal-source-format/8233/"><strike>test-portal-source-format</strike></a></li></ul></div><h5>For more details click <a href="https://test-1-37.liferay.com/userContent/jobs/test-portal-source-format/builds/8233/jenkins-report.html">here</a>.</h5><pre>     [java] java.lang.Exception: Found 4 formatting issues:
     [java] 1: Variable 'Item' in 'ValidateConfigurationStringAtControlPanel' should start with a lowercase letter: /opt/dev/projects/github/liferay-portal/portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/testcase/MyTestFile.testcase (SourceCheck:PoshiVariableNameCheck)
     [java] 2: /opt/dev/projects/github/liferay-portal/portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/Path/FileTest.path expected:&lt;...d&gt;
     [java]   &lt;td&gt;&lt;/td&gt;
     [java] &lt;/tr&gt;
     [java] [&lt;tr&gt;
     [java]   &lt;td&gt;CLICK_CTP&lt;/td&gt;
     [java]   &lt;td&gt;//*[@id="_com_liferay_product_navigation_applications_menu_web_internal_portlet_ProductNavigationApplicationsMenuPortlet_tab_2"]/button&lt;/td&gt;
     [java]   &lt;td&gt;&lt;/td&gt;
     [java] &lt;/tr&gt;
     [java] &lt;tr&gt;
     [java]   &lt;td&gt;CHECK_CONF&lt;/td&gt;
     [java]   &lt;td&gt;//*[@id="nav_control_panel.configuration"]&lt;/td&gt;
     [java]   &lt;td&gt;&lt;/td&gt;
     [java] &lt;/tr&gt;
     [java] 
     [java] &lt;!-- Part of Screen size --&gt;
     [java] 
     [java] &lt;tr&gt;
     [java]   &lt;td&gt;CLICK_SIMULATION&lt;/td&gt;
     [java]   &lt;td&gt;//*[@id="_com_liferay_product_navigation_simulation_web_portlet_SimulationPortlet_simulationToggleId"]&lt;/td&gt;
     [java]   &lt;td&gt;&lt;/td&gt;
     [java] &lt;/tr&gt;
     [java] &lt;tr&gt;
     [java]   &lt;td&gt;CLICK_CUSTOM&lt;/td&gt;
     [java]   &lt;td&gt;//*[@id="_com_liferay_product_navigation_simulation_web_portlet_SimulationPortlet_simulationDeviceContainer"]/div/div[1]/button[5]&lt;/td&gt;
     [java]   &lt;td&gt;&lt;/td&gt;
     [java] &lt;/tr&gt;
     [java] &lt;tr&gt;
     [java]   &lt;td&gt;INPUT_WIDTH&lt;/td&gt;
     [java]   &lt;td&gt;//*[@id="_com_liferay_product_navigation_simulation_web_portlet_SimulationPortlet_width"]&lt;/td&gt;
     [java]   &lt;td&gt;&lt;/td&gt;
     [java] &lt;/tr&gt;
     [java] &lt;tr&gt;
     [java]   &lt;td&gt;INPUT_HEIGHT&lt;/td&gt;
     [java]   &lt;td&gt;//*[@id="_com_liferay_product_navigation_simulation_web_portlet_SimulationPortlet_height"]&lt;/td&gt;
     [java]   &lt;td&gt;&lt;/td&gt;
     [java] &lt;/tr&gt;]
     [java] 
     [java] &lt;/tbody&gt;
     [java] &lt;/table&gt;
     [java] ...&gt; but was:&lt;...d&gt;
     [java]   &lt;td&gt;&lt;/td&gt;
     [java] &lt;/tr&gt;
     [java] [
     [java] 
     [java] &lt;tr&gt;
     [java]   &lt;td&gt;CLICK_CTP&lt;/td&gt;
     [java]   &lt;td&gt;//*[@id="_com_liferay_product_navigation_applications_menu_web_internal_portlet_ProductNavigationApplicationsMenuPortlet_tab_2"]/button&lt;/td&gt;
     [java]   &lt;td&gt;&lt;/td&gt;
     [java] &lt;/tr&gt;
     [java] 
     [java] 
     [java] 
     [java] &lt;tr&gt;
     [java]   &lt;td&gt;CHECK_CONF&lt;/td&gt;
     [java]   &lt;td&gt;//*[@id="nav_control_panel.configuration"]&lt;/td&gt;
     [java]   &lt;td&gt;&lt;/td&gt;
     [java] &lt;/tr&gt;</pre></details></html>

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-37.liferay.com/job/test-portal-source-format/8233/">test-portal-source-format#8233</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-37.liferay.com/userContent/jobs/test-portal-source-format/builds/8233/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>sf<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/pedroamarall/liferay-portal/pull/13">pedroamarall#13</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/builds?testrayRoutineId=1306753894">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/runs?testrayBuildId=2428926833">[master] ci:test:sf - Youness-ELMO &gt; pedroamarall - PR#13 - 2023-09-21[09:45:31]</a><br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-37.liferay.com/job/publish-testray-report/31259/">publish-testray-report#31259</a><br/></div></div>

pedroamarall commented:
ci:reopen

pedroamarall commented:
ci:help

liferay-continuous-integration commented:
## Available CI commands:
#### ci:close
&nbsp;&nbsp;&nbsp;&nbsp;Close the pull request.
#### ci:forward[:force]
&nbsp;&nbsp;&nbsp;&nbsp;Test the pull request  and forward the pull request to `brianchandotcom` if the required test suites pass for `ci:forward` or complete for `ci:forward:force`.
#### ci:merge[:force]
&nbsp;&nbsp;&nbsp;&nbsp;Merge in the changes from the subrepo. All tests must pass before this command will successfully run. Optionally use the force flag to bypass failed tests.
#### ci:reevaluate:[buildID]
&nbsp;&nbsp;&nbsp;&nbsp;Reevaluate the pull request result from a generated build ID.
#### ci:reopen
&nbsp;&nbsp;&nbsp;&nbsp;Reopen the pull request.
#### ci:stop[:suite]
&nbsp;&nbsp;&nbsp;&nbsp;Stop all currrently running tests. Optionally specify the name of the test suite.
#### ci:test[:suite][:SHA|nocompile|norebase|gist-SHA]
&nbsp;&nbsp;&nbsp;&nbsp;Test the pull request. Optionally specify the name of the test suite and/or optionally specify the upstream SHA to test against or use "nocompile" to test with a prebuilt bundle or use "norebase" to test without rebasing.
<details><summary><strong>Available test suites:</strong></summary>

- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**acceptance** - Test acceptance tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**acceptance-functional** - Test acceptance functional tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**acceptance-functional-db-partition** - Test acceptance functional tests with Database Partitioning.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**acceptance-release** - Compile a release bundle and execute acceptance tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**acceptance-upstream** - Test acceptance upstream tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**accessibility** - Test all accessibility tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**accessibility-menu** - Test all accessibility menu tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**account** - Test Account tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**analytics-cloud** - Test Analytics Cloud tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**analytics-cloud-acceptance** - No description is available.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**analytics-cloud-integration** - No description is available.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**analytics-cloud-quarantine** - No description is available.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**app-servers** - Test Deployments on App Servers.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**auto-tagging** - Test Auto Tagging tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**backend** - Test backend tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**backend-hypersonic** - Test backend tests on Hypersonic.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**blade** - Test Blade samples tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**blogs** - Test Blogs tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**bulk-editing** - Test Bulk Editing tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**bundle** - Saves a compiled tomcat bundle.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**calendar** - Test Calendar tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**clay** - No description is available.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**click-to-chat** - Test Click to Chat tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**clustering** - Test clustering tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**comment** - Test Comment tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**commerce** - Test only Commerce tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**commerce-database** - Test only Commerce Database tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**commerce-es7-remote** - Test only Commerce tests with the latest compatible remote Elasticsearch 7 version.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**commerce-es8-remote** - Test only Commerce tests with the latest compatible remote Elasticsearch 8 version.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**commerce-functional** - Test only Commerce functional tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**commerce-integration** - Test only Commerce integration tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**commerce-upgrade** - No description is available.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**content-page-review** - Test Content Page Review (comments) tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**continuous-integration** - Test Continuous Integration tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**cookies** - Test Cookie Preferences tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**data-engine** - Test Data Engine functional tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**data-engine-db-partition** - Test Data Engine functional tests with Database Partitioning.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**data-integration** - Test only Data Integration tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**data-integration-functional** - Test only functional Data Integration tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**db-partition** - Test Database Partitioning functional tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**depot** - Test Depot tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**digital-signature** - Test Digital Signature tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**dl-store** - Test Document Library Store tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**document** - Test Document and Media tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**document-antivirus** - Test Antivirus against DM.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**drools** - Test Drools tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**dummy** - No description is available.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**echo** - Test all Echo (WCM) team tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**echo-acceptance** - Test Echo (WCM) team acceptance tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**echo-accessibility** - Test Echo (WCM) team accessibility tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**echo-analytics-cloud** - Test segmentation and personalization with Analytics Cloud.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**echo-db-partition** - Test Echo (WCM) team acceptance tests with Database Partitioning.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**echo-headless** - Test Echo (WCM) team headless components.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**echo-upgrade** - Test Echo (WCM) team upgrade tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**environments** - Test environment acceptance functional tests on all available environment stacks.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**export-import** - Test exportimport tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**forms** - Test forms tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**forms-db-partition** - Test forms functional tests with Database Partitioning.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**forms-functional** - Test only functional forms tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**friendly-url** - Test Friendly URL tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**frontend** - Test all Frontend Infra team tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**frontend-client-extensions** - Test frontend client extensions tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**frontend-data-set** - Test frontend data set tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**frontend-dev-exp** - Test frontend developer experience tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**frontend-ui** - Test frontend ui tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**frontend-walkthrough** - Test frontend walkthrough tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**gauntlet** - Test upstream tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**gauntlet-bucket** - Test a subset of upstream tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**gradle-plugins** - Test Gradle Plugins tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**headless** - Test all headless team tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**headless-acceptance** - Test headless team acceptance tests considering Headless and Rest Builder.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**headless-builder** - All tests for Headless Builder component.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**headless-functional** - Headless Team functional API tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**hotel** - Test all Source Formatter unit tests related to upgrade
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**item-selector** - Test item selector tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**jacoco-report** - Test integration and unit tests to generate JaCoCo reports.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**jdk11** - Test acceptance, smoke, and upgrade functional tests against JDK11.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**jsf** - Test Liferay Faces tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**kaleo-forms** - Test Kaleo Forms admin tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**knowledge-base** - No description is available.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**ldap** - Test LDAP tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**license** - Compile a release bundle and execute tests for licensing.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**license-cluster** - Compile a release bundle and execute tests for cluster licensing on 32GB slaves.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**liferay-online** - Compile a release bundle and execute tests with liferay online properties.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**liferay-sample-workspace** - No description is available.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**lima** - Test Lima tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**lima-acceptance** - Test Lima acceptance tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**lima-db-partition** - Test Lima functional tests with Database Partitioning.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**lima-headless** - Test Lima headless tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**lima-lxc** - Lima functional tests with Liferay Experience Cloud configurations.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**lima-upgrade** - Test Lima upgrade tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**lpkg** - Test LPKG tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**message-boards** - No description is available.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**mfa** - Test Multi Factor Authentication tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**notifications** - Test Notifications tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**oauth2** - Test OAuth 2 tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**object** - Test Object tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**object-environments** - Test Object tests on available environment stacks.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**object-lxc** - Test Object tests with Liferay Experience Cloud configurations.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**oidc** - Test OpenID Connect tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**onedrive** - Test OneDrive tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**online-editing** - Test Online Editing tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**osb-site-initializer-customer-portal** - No description is available.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**osb-site-initializer-evp** - No description is available.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**osb-site-initializer-partner-portal** - No description is available.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**password-policies** - Test Password Policies tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**peds** - No description is available.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**peds-acceptance** - No description is available.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**peds-accessibility** - No description is available.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**peds-sm** - No description is available.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**portal-context-path** - No description is available.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**portal-proxy-path** - No description is available.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**portlet-deployment** - Test deployment of all demo portlet plugins.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**poshi** - No description is available.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**poshi-gauntlet** - No description is available.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**project-templates** - Test project-templates tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**publications** - Test publications tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**publications-as** - Test publications tests with different application servers.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**publications-db** - Test publications tests with different databases.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**publications-db-partition** - Test publications functional tests with Database Partitioning.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**publications-lxc** - Test publications tests with Liferay Experience Cloud configurations.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**questions** - No description is available.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**redirect** - Test Redirection tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**relevant** - Test relevant changes only.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**reports** - Test Reports tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**roles** - Test Roles tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**saml** - Test SAML tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**search** - Test search tests with the Elasticsearch 7 search engine.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**search-as** - Test search tests with the Elasticsearch 7 search engine on different application servers.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**search-db** - Test search tests with the Elasticsearch 7 search engine on different databases.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**search-db-partition** - Test search functional tests with Database Partitioning.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**search-es8-as** - Test search tests with the Elasticsearch 8 search engine on different application servers.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**search-es8-db** - Test search tests with the Elasticsearch 8 search engine on different databases.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**search-es8-remote** - Test only search tests with the latest compatible remote Elasticsearch 8 version.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**search-es8-remote-minimum** - Test search tests with the minimum compatible remote Elasticsearch 8 version.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**search-lxc** - Test search tests with the Elasticsearch 7 search engine and Liferay Experience Cloud configurations.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**search-remote** - Test only search tests with the latest compatible remote Elasticsearch 7 version.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**search-remote-minimum** - Test search tests with the minimum compatible remote Elasticsearch 7 version.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**search-solr** - Test search tests with the Solr 8 search engine.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**search-solr-as** - Test search tests with the Solr 8 search engine on different application servers.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**search-solr-db** - Test search tests with the Solr 8 search engine on different databases.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**security** - Test security tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**seo** - Test SEO tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**service-builder** - Test Service Builder tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**setup-wizard** - Test Setup Wizard tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**sf** - Test source formatter.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**sharepoint** - Test Sharepoint SOAP 2013 and REST tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**sharing** - Test Asset Sharing tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**site-initializer-liferay-marketplace** - No description is available.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**site-initializer-liferaybotics** - No description is available.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**site-initializer-masterclass** - No description is available.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**site-initializer-raylife-ap** - No description is available.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**site-initializer-raylife-d2c** - No description is available.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**site-initializer-team-extranet** - No description is available.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**site-initializer-testray** - No description is available.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**site-template** - Test Site Template tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**smoke** - Run smoke test with all available configurations in PortalSmoke.testcase
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**solutions-acceptance** - Test Solutions acceptance tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**stable** - No description is available.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**staging** - Test Staging components tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**staging-custom-db** - No description is available.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**startup** - Test Portal startup.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**sybase** - Test functional and integration tests on Sybase database.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**system** - Test functional and integration tests for System Management (Virtual Instances, System Settings, Instance Settings)
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**tck** - Test TCK tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**themes** - No description is available.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**translation** - Test Translation Management tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**um** - Test User Management tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**um-db-partition** - No description is available.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**upgrade** - Test upgrades against tomcat app server and all supported databases.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**upgrade-all** - Test upgrades from all components against all app servers and databases.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**upgrade-app-servers** - Test upgrades against all app servers.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**upgrade-client** - Test upgrade client.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**upgrade-performance** - Test large database for upgrade performance regressions.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**upstream-functional** - Test upstream functional tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**upstream-functional-db-partition** - Test upstream functional tests with Database Partitioning.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**users** - Test Users and Organizations tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**usm** - No description is available.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**usm-db-partition** - No description is available.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**usm-environments** - No description is available.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**usm-upgrade** - No description is available.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**vulcan-acceptance** - Test headless team acceptance tests considering Vulcan and Commerce.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**wiki** - Test wiki tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**workflow** - Test Workflow tests.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**workflow-db-partition** - Test Workflow functional tests with Database Partitioning.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**workflow-es7-remote** - Test Workflow tests with the latest compatible remote Elasticsearch 7 version.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**workflow-es8-remote** - Test Workflow tests with the latest compatible remote Elasticsearch 8 version.
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**workflow-metrics** - Test Workflow metrics tests.
</details>

For more details, see <a href="https://liferay.atlassian.net/wiki/spaces/SUPPORT/pages/1957036035/CI+liferay-continuous-integration+GitHub+Commands">CI GitHub Commands</a>

Youness-ELMO commented on portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/macro/TestTraining.macro:
```diff
@@ -0,0 +1,204 @@
+        );
+    }
+
+
```
Done

Youness-ELMO commented on portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/macro/TestTraining.macro:
```diff
@@ -0,0 +1,204 @@
+definition {
+
+ //Ex 1
+    macro ClickMenu {
```
Done

Youness-ELMO commented on portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/macro/TestTraining.macro:
```diff
@@ -0,0 +1,204 @@
+    }
+
+
+    macro ClickSimulation {
```
Fixed, thanks for the comment

Youness-ELMO commented on portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/macro/TestTraining.macro:
```diff
@@ -0,0 +1,204 @@
+        Click(
+            locator1 = "FileTest#CLICK_MENU"
+        );
+    }
```
done

Youness-ELMO commented on portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/macro/TestTraining.macro:
```diff
@@ -0,0 +1,204 @@
+        Click(
+            locator1 = "FileTest#CLICK_CUSTOM"
+        );
+    }
```
done

Youness-ELMO commented on portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/macro/TestTraining.macro:
```diff
@@ -0,0 +1,204 @@
+    }
+
+
+    // Ex 2: Part of form
```
thanks, its done

Youness-ELMO commented on portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/macro/TestTraining.macro:
```diff
@@ -0,0 +1,204 @@
+
+
+
+    }
```
oops, I forget, let me fix it, sorry!

Youness-ELMO commented on portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/macro/TestTraining.macro:
```diff
@@ -0,0 +1,204 @@
+    }
+
+
+// Extra exercise : Blog Part
```
done

Youness-ELMO commented on portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/testcase/MyTestFile.testcase:
```diff
@@ -0,0 +1,138 @@
+    //     User.logoutAndLoginPG(
+    //         userLoginEmailAddress = "test@liferay.com",
+    //         userLoginFullName = "Test Test");
+    // }
```
oky i fixe it

Youness-ELMO commented on portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/macro/TestTraining.macro:
```diff
@@ -0,0 +1,204 @@
+
+
+
+}
```
done

Youness-ELMO commented on portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/testcase/MyTestFile.testcase:
```diff
@@ -0,0 +1,138 @@
+                entrySearchTerm = "This is "
+            );
+        }
+    }
```
ohhh thank you , yes i change the task description

Youness-ELMO commented on portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/testcase/MyTestFile.testcase:
```diff
@@ -0,0 +1,138 @@
+        User.firstLoginPG();
+
+        Navigator.openURL();
+    }
```
when i do the formating, he don't delete the spaces so i need to fix it manually !!

pedroamarall commented:
ci:test:sf

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>sf</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-40.liferay.com/job/test-portal-source-format/2993/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-1-40.liferay.com/userContent/jobs/test-portal-source-format/builds/2993/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:heavy_check_mark: ci:test:sf - 1 out of 1 jobs passed in 4 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/33dcc7116256d2c52b3a676d48e7ed3276320ed4">33dcc7116256d2c52b3a676d48e7ed3276320ed4</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/Youness-ELMO/liferay-portal/tree/LPS-12345-Poshi-exercises">LPS-12345-Poshi-exercises</a><br/>Branch GIT ID: <a href="https://github.com/Youness-ELMO/liferay-portal/commit/3eb036238cfa49ea81d61e55892b0dade271626e">3eb036238cfa49ea81d61e55892b0dade271626e</a></p>1 out of 1jobs PASSED<details><summary><strong>1 Successful Jobs:</strong></summary><ul><li><a href="https://test-1-40.liferay.com/job/test-portal-source-format/2993/">test-portal-source-format</a></li></ul></details><h5>For more details click <a href="https://test-1-40.liferay.com/userContent/jobs/test-portal-source-format/builds/2993/jenkins-report.html">here</a>.</h5></details></html>

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-40.liferay.com/job/test-portal-source-format/2993/">test-portal-source-format#2993</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-40.liferay.com/userContent/jobs/test-portal-source-format/builds/2993/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>sf<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/pedroamarall/liferay-portal/pull/13">pedroamarall#13</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/builds?testrayRoutineId=1306753894">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/runs?testrayBuildId=2430291726">[master] ci:test:sf - Youness-ELMO &gt; pedroamarall - PR#13 - 2023-09-25[02:44:19]</a><br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-40.liferay.com/job/publish-testray-report/11577/">publish-testray-report#11577</a><br/></div></div>

pedroamarall commented on portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/macro/TestTraining.macro:
```diff
@@ -0,0 +1,162 @@
+		ApplicationsMenuHelper.openApplicationsMenu();
+	}
+
+	macro clickOnForm {
```
A more appropriate name would be "CreateForm" because this macro is not just clicking, its function is to create a form, correct?

Youness-ELMO replied:
done

Youness-ELMO replied:
Done

pedroamarall commented on portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/testcase/MyTestFile.testcase:
```diff
@@ -0,0 +1,129 @@
+	@description = "Access Portal Global Menu and validate CONFIGURATION string at Control Panel"
+	@priority = 5
+	test ValidateConfigurationStringAtControlPanel {
+		task ("Go to menu then Control Panel then check if CONFIGURATION its exist or not") {
+			TestTraining.ClickMenu();
+
+			TestTraining.ClickControlPanel();
+
+			TestTraining.VerifyCONF(item = "CONFIGURATION");
```
Use BDD correctly please (GIVEN, WHEN and THEN)

Youness-ELMO replied:
Done

pedroamarall commented on portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/macro/TestTraining.macro:
```diff
@@ -0,0 +1,162 @@
+		Click(locator1 = "FileTest#CLICK_CTP");
+	}
+
+	macro ClickMenu {
+		ApplicationsMenuHelper.openApplicationsMenu();
```
Why did you create a macro that already does this?

Youness-ELMO replied:
done

pedroamarall commented on portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/testcase/MyTestFile.testcase:
```diff
@@ -0,0 +1,129 @@
+		TestTraining.clickOnForm();
+	}
+
+	test Ctrlpanel {
+		task ("blablablaaa") {
+			TestTraining.ClickMenu();
+		}
+	}
```
Your test is just 1 macro?

Youness-ELMO replied:
sorry for this one just a test i fix it, thank you

pedroamarall commented on portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/testcase/MyTestFile.testcase:
```diff
@@ -0,0 +1,129 @@
+
+	@description = "Add some fields in form part"
+	@priority = 5
+	test CreateForm {
+		TestTraining.clickOnForm();
+	}
```
You could divide this macro into several parts since I saw that it performs several functions, as you can use it in other tests and you can also organize your test case with BDD

Youness-ELMO replied:
safi its done, i split it to many tasks

manoelcyreno commented:
starting the review :eyes:

manoelcyreno commented on portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/Path/FileTest.path:
```diff
@@ -0,0 +1,53 @@
+</tr>
+<tr>
+	<td>CLICK_CUSTOM</td>
+	<td>//*[@id="_com_liferay_product_navigation_simulation_web_portlet_SimulationPortlet_simulationDeviceContainer"]/div/div[1]/button[5]</td>
```
This locator is using a mix between an absolute and relative XPath, please, change it to use the **relative xpath** in the right way

Youness-ELMO replied:
Done

manoelcyreno commented on portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/macro/TestTraining.macro:
```diff
@@ -0,0 +1,157 @@
+
+		FormViewBuilder.clickFieldLabel(fieldLabel = "Age");
+
+		Form.editElementSetFieldLabel(fieldValue = "âge");
+		FormViewBuilder.clickFieldLabel(fieldLabel = "Age");
```
add a new line between those lines

Youness-ELMO replied:
Done

manoelcyreno commented on portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/macro/TestTraining.macro:
```diff
@@ -0,0 +1,157 @@
+
+		FormFields.editText(
+			fieldName = "Numeric",
+			fieldValue = 21);
+		FormFields.editText(
```
add a new line between those lines

Youness-ELMO replied:
Done

manoelcyreno commented on portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/macro/TestTraining.macro:
```diff
@@ -0,0 +1,157 @@
+
+		FormFields.editText(
+			fieldName = "Numeric",
+			fieldValue = 21);
+		FormFields.editText(
```
add a new line between those lines

Youness-ELMO replied:
Done

manoelcyreno commented on portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/testcase/MyTestFile.testcase:
```diff
@@ -0,0 +1,216 @@
+@component-name = "portal-web-forms-and-data-lists"
+definition {
+
+	property testray.main.component.name = "Forms";
+
+	property testray.main.component.name = "Forms";
```
duplicated lines.

Youness-ELMO replied:
sorry for those duplicated lines, yesterday I merged some files and it showed me these duplicate lines, I fixed the duplicate lines, thank you!

manoelcyreno commented on portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/testcase/MyTestFile.testcase:
```diff
@@ -0,0 +1,216 @@
+
+	property testray.main.component.name = "Forms";
+
+	setUp {
+		TestCase.setUpPortalInstance();
+	setUp {
+		TestCase.setUpPortalInstance();
```
duplicated lines.

Youness-ELMO replied:
Done

manoelcyreno commented on portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/testcase/MyTestFile.testcase:
```diff
@@ -0,0 +1,216 @@
+	setUp {
+		TestCase.setUpPortalInstance();
+
+		User.firstLoginPG();
+		User.firstLoginPG();
```
duplicated lines.

Youness-ELMO replied:
Done

manoelcyreno commented on portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/testcase/MyTestFile.testcase:
```diff
@@ -0,0 +1,216 @@
+		User.firstLoginPG();
+		User.firstLoginPG();
+
+		Navigator.openURL();
+	}
+		Navigator.openURL();
+	}
```
duplicated lines.

Youness-ELMO replied:
Done

manoelcyreno commented on portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/testcase/MyTestFile.testcase:
```diff
@@ -0,0 +1,216 @@
+		Navigator.openURL();
+	}
+
+	tearDown {
+	tearDown {
```
duplicated lines.

Youness-ELMO replied:
Done

manoelcyreno commented on portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/testcase/MyTestFile.testcase:
```diff
@@ -0,0 +1,216 @@
+	tearDown {
+		var testPortalInstance = PropsUtil.get("test.portal.instance");
+
+		if (${testPortalInstance} == "true") {
+		if (${testPortalInstance} == "true") {
```
duplicated lines.

Youness-ELMO replied:
Done

manoelcyreno commented on portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/testcase/MyTestFile.testcase:
```diff
@@ -0,0 +1,216 @@
+		if (${testPortalInstance} == "true") {
+			PortalInstances.tearDownCP();
+		}
+	}
+	}
```
duplicated lines.

Youness-ELMO replied:
Done

manoelcyreno commented on portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/testcase/MyTestFile.testcase:
```diff
@@ -0,0 +1,216 @@
+			BlogsNavigator.openBlogsAdmin(siteURLKey = "guest");
+		}
+
+		task ("When created new Blog") {
+			TestTraining.CreateBlogs(
+				entryContent = "This is Blog Content",
+				entryTitle = "This is Title");
+		}
```
Replace it with a JSON call about creating a blog.

Youness-ELMO replied:
Done, it a new information thank you!

manoelcyreno commented on portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/testcase/MyTestFile.testcase:
```diff
@@ -0,0 +1,216 @@
+			TestTraining.DeleteBlog(rowEntry = "This is Title");
+		}
+	}
+	@description = "Can Create a Blog"
+	@priority = 5
+	test CanCreateBlog {
+		task ("Given Go to Blogs Part") {
+			BlogsNavigator.openBlogsAdmin(siteURLKey = "guest");
+		}
+
+		task ("Then created new Blog") {
+			TestTraining.CreateBlogs(
+				entryContent = "This is Blog Content",
+				entryTitle = "This is Title");
+		}
+	}
```
duplicated code

Youness-ELMO replied:
Done

manoelcyreno commented on portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/testcase/MyTestFile.testcase:
```diff
@@ -0,0 +1,216 @@
+		}
+	}
+
+	@description = "Can Delete a Blog"
+	@priority = 5
+	test CanDeleteBlog {
+		task ("Given Go to Blogs Part") {
+			BlogsNavigator.openBlogsAdmin(siteURLKey = "guest");
+		}
+
+		task ("When created new Blog") {
+			TestTraining.CreateBlogs(
+				entryContent = "This is Blog Content",
+				entryTitle = "This is Title");
+		}
+
+		task ("Then delete the Blog contain this title") {
+			TestTraining.DeleteBlog(rowEntry = "This is Title");
+		}
+	}
```
duplicated code

Youness-ELMO replied:
fixed, thank you!

manoelcyreno commented on portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/testcase/MyTestFile.testcase:
```diff
@@ -0,0 +1,216 @@
+			BlogsNavigator.openBlogsAdmin(siteURLKey = "guest");
+		}
+
+		task ("Given created new Blog") {
+			TestTraining.CreateBlogs(
+				entryContent = "This is Blog Content",
+				entryTitle = "This is Title");
```
Replace it with a JSON call about creating a blog.

Youness-ELMO replied:
Done

manoelcyreno commented on portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/testcase/MyTestFile.testcase:
```diff
@@ -0,0 +1,216 @@
+			BlogsNavigator.openBlogsAdmin(siteURLKey = "guest");
+		}
+
+		task ("When created new Blog") {
+			TestTraining.CreateBlogs(
+				entryContent = "This is Blog Content",
+				entryTitle = "This is Title");
+		}
```
Replace it with a JSON call about creating a blog.

Youness-ELMO replied:
done

manoelcyreno commented:
@Youness-ELMO, take a look at:
1- all "Given", "When", "Then", and "And" that you created, in some cases, are in lower cases, in other all character is in higher case...
2- The name of your methods, and the "Then", do not pass to me what in fact it wants to validate... so, please, change it.
3- Review all the comments above.

manoelcyreno reviewed (changes requested):
request changes on the PR

Youness-ELMO commented:
> @Youness-ELMO, take a look at: 1- all "Given", "When", "Then", and "And" that you created, in some cases, are in lower cases, in other all character is in higher case... 2- The name of your methods, and the "Then", do not pass to me what in fact it wants to validate... so, please, change it. 3- Your code is about Blogs or Forms? in only 1 testcase file, you added test code, about these 2 different components/features... split it. 4- Review all the comments above.

@manoelcyreno  Thank you for your comments, actually I fixed almost all of those errors, the problem is I didn't understand the second request [2- The name of your methods, and the "Then", do not pass to me what in fact it wants to validate... so, please, change it."], what i need to do in "Then" tasks exactly, thanks for your time, have a great day!

manoelcyreno commented on portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/testcase/MyTestFile.testcase:
```diff
@@ -0,0 +1,48 @@
+	@description = "Access Portal Global Menu and validate CONFIGURATION string at Control Panel"
+	@priority = 5
+	test ValidateConfigurationStringAtControlPanel {
+		task ("GIVEN Go to Menu") {
+			ApplicationsMenuHelper.openApplicationsMenu();
+		}
+
+		task ("WHEN Control Panel ") {
+			TestTraining.ClickControlPanel();
+		}
+
+		task ("then check if Confirmation its exist or not") {
+			TestTraining.VerifyCONF(item = "CONFIGURATION");
+		}
```
GIVEN / WHEN / then

Is this look right? this is following some pattern?

manoelcyreno commented on portal-web/test/functional/com/liferay/portalweb/tests/myTestPoshi/testcase/MyTestFile.testcase:
```diff
@@ -0,0 +1,48 @@
+		}
+
+		task ("then check if Confirmation its exist or not") {
+			TestTraining.VerifyCONF(item = "CONFIGURATION");
```
Why do you create a macro for it?  Why didn't you add the assert here?

manoelcyreno commented:
@Youness-ELMO, take a look at:
1- all "Given", "When", "Then", and "And" that you created, in some cases, are in lower cases, in other all character is in higher case...
2- The name of your methods, and the "Then", do not pass to me what it wants to validate... so, please, change it.
3- Review all the comments above.


example about the point 2.

> "Then Verify the results"
Verify what? fail situation? success situation?

> test CreateForm
Create with success? with failure? is not possible to create? is possible?

Think about a person without any context who will read it and this person will understand what you want to test, without reading your code in details?
